packages/web-components/src/components/va-pagination/va-pagination.tsx

```diff
@@ -338,71 +391,137 @@ export class VaPagination {
+            }
+            onKeyDown={e => this.handleKeyDown(e, this.pages)}
+            href="javascript:void(0)"
+            class="usa-pagination__button"
+            aria-label={`page ${this.pages}, last page`}
+          >
+            {this.pages}
+          </a>
+        </li>
+      );
+    }
+
+    return (
+      <Fragment>
+        {lastEllipsis}
+        {lastPageButton}
+      </Fragment>
+    );
+  }
+
+  private renderNextPageButton(
+    arrowClasses: string,
+    nextAriaLabel: string,
+  ): JSX.Element {
+    // Skip rendering if the viewport is too narrow to show Next button
+    if (this.isMobileViewport) return null;
+
+    // Return early if we're on the last page or current page was entered incorrectly
+    if (this.page >= this.pages) return null;
+
+    return (
+      <li class={arrowClasses}>
+        <a
+          aria-label={nextAriaLabel}
+          onClick={() =>
+            this.handlePageSelect(this.page + 1, 'nav-paginate-number')
+          }
+          onKeyDown={e => this.handleKeyDown(e, this.page + 1)}
+          class="usa-pagination__link usa-pagination__next-page"
+          href="javascript:void(0)"
+        >
+          <span class="usa-pagination__link-text">{i18next.t('next')}</span>
+          <div id="next-arrow-icon"></div>
+        </a>
+      </li>
+    );
+  }
+  /**
+   * =========================================
+   * End pagination render functions
+   * =========================================
+   */
+
+  render() {
+    const { ariaLabelSuffix, pages } = this;
+
+    // Return early if there's no need to render pagination
+    if (pages === 1) return null;
+
+    const previousAriaLabel = ariaLabelSuffix
+      ? `Previous page ${ariaLabelSuffix}`
+      : 'Previous page';
+
+    const nextAriaLabel = ariaLabelSuffix
+      ? `Next page ${ariaLabelSuffix}`
+      : 'Next page';
+
+    const itemClasses = classnames({
+      'usa-pagination__item': true,
+      'usa-pagination__page-no': true,
+      'va-pagination__item': true,
+    });
+
+    const ellipsisClasses = classnames({
+      'usa-pagination__item': true,
+      'usa-pagination__overflow': true,
+      'va-pagination__item': true,
+      'va-pagination__overflow--isMobile': this.isMobileViewport,
+    });
+
+    const arrowClasses = classnames({
+      'usa-pagination__item': true,
+      'usa-pagination__arrow': true,
+    });
 
     return (
       <Host>
         <nav class="usa-pagination" aria-label="Pagination">
           <ul class="usa-pagination__list">
-            {previousButton}
-            {renderPages}
-            {endEllipsisAndLastPage}
-            {nextButton}
+            {this.renderPreviousButton(arrowClasses, previousAriaLabel)}
+            {this.renderFirstPage(itemClasses, ellipsisClasses)}
+            {this.renderMiddlePages(ariaLabelSuffix, itemClasses)}
+            {this.renderLastPage(ellipsisClasses, itemClasses)}
+            {this.renderNextPageButton(arrowClasses, nextAriaLabel)}
```

Comment:
I re-architected these into private functions so we could separate the business logic from the `render()` method. I felt it made the "view layer" easier to read and co-located each piece of the pagination. If there's a better way to render these than individual functions, LMK!